vql/tools/humio/plugin.go

```diff
@@ -0,0 +1,256 @@
+					args.HttpTimeoutSec),
+		}
+	}
+
+	return nil
+}
+
+func applyArgs(args *humioPluginArgs, queue *HumioQueue) error {
+	var err error
+	if args.Threads > 0 {
+		err = queue.SetWorkerCount(args.Threads)
+		if err != nil {
+			return fmt.Errorf("`threads': %w", err)
+		}
+	}
+
+	if args.BatchingTimeoutMs > 0 {
+		timeout := time.Duration(args.BatchingTimeoutMs) * time.Millisecond
+		err = queue.SetBatchingTimeoutDuration(timeout)
+		if err != nil {
+			return fmt.Errorf("`batching_timeout_ms': %w", err)
+		}
+	}
+
+	if args.EventBatchSize > 0 {
+		err = queue.SetEventBatchSize(args.EventBatchSize)
+		if err != nil {
+			return fmt.Errorf("`event_batch_size': %w", err)
+		}
+	}
+
+	if args.HttpTimeoutSec > 0 {
+		timeout := time.Duration(args.HttpTimeoutSec) * time.Second
+		err = queue.SetHttpClientTimeoutDuration(timeout)
+		if err != nil {
+			return fmt.Errorf("`http_timeout': %w", err)
+		}
+	}
+
+	if args.MaxRetries != defaultMaxRetries {
+		err = queue.SetMaxRetries(args.MaxRetries)
+		if err != nil {
+			return fmt.Errorf("`max_retries': %s", err)
+		}
+	}
+
+	err = queue.SetTaggedFields(args.TagFields)
+	if err != nil {
+		return fmt.Errorf("`tag_fields': %w", err)
+	}
+
+	if args.Debug {
+		queue.EnableDebugging(true)
+	}
+
+	return nil
+}
+
+func (self humioPlugin) Call(ctx context.Context,
+	scope vfilter.Scope,
+	args *ordereddict.Dict) <-chan vfilter.Row {
+	outputChan := make(chan vfilter.Row)
+
+	go func() {
+		defer close(outputChan)
+
+		err := vql_subsystem.CheckAccess(scope, acls.COLLECT_SERVER)
+		if err != nil {
+			scope.Log("humio: %v", err)
+			return
+		}
+
+		arg := humioPluginArgs{
+			MaxRetries: defaultMaxRetries,
+		}
+		err = arg_parser.ExtractArgsWithContext(ctx, scope, args, &arg)
+		if err != nil {
+			scope.Log("humio: %v", err)
+			return
+		}
+
+		err = arg.validate()
+		if err != nil {
+			scope.Log("humio: %v", err)
+			return
+		}
+
+		config_obj, ok := vql_subsystem.GetServerConfig(scope)
+		if !ok {
+			scope.Log("humio: could not get config from scope")
+			return
+		}
+
+		queue := NewHumioQueue(config_obj)
+
+		err = applyArgs(&arg, queue)
+		if err != nil {
+			scope.Log("humio: %v", err)
+			return
+		}
+
+		transport, err := networking.GetHttpTransport(config_obj.Client, arg.RootCerts)
+		if err != nil {
+			scope.Log("humio: %v", err)
+			return
+		}
+
+		if arg.SkipVerify {
+			err = networking.EnableSkipVerify(transport.TLSClientConfig,
+							  config_obj.Client)
+			if err != nil {
+				scope.Log("humio: %v", err)
+				return
+			}
+		}
+
+		err = queue.SetHttpTransport(transport)
+		if err != nil {
+			scope.Log("humio: %v", err)
+			return
+		}
+
+		err = queue.Open(scope, arg.ApiBaseUrl, arg.IngestToken)
+		if err != nil {
+			scope.Log("humio: could not open queue: %v", err)
+			return
+		}
+		queue.Log(scope, "plugin started (%v threads)", queue.WorkerCount())
+		defer queue.Log(scope, "plugin terminating")
+		defer queue.Close(scope)
```

Comment:
I think this will wait until the queue is fully shutdown? the Close() method has a wg.Wait() no?

Reply:
It will. Should it not?